File: PHP/TeoremaDaDivisaoEuclidianaTest.php
<?php
/**
 * @version PHP
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit NomeDaClasseTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("TeoremaDaDivisaoEuclidiana.php");

class TeoremaDaDivisaoEuclidianaTest extends TestCase {

    public function testTeoremaDaDivisaoEuclidiana() {
        $obj = new TeoremaDaDivisaoEuclidiana();
        //Test 1
        $want = [2, 1];
        $got = $obj->getQuoAndRest( 7, 3 );
        $this->assertEquals($want, $got);
        //Test 2
        $want = [-2, 1];
        $got = $obj->getQuoAndRest( 7, -3 );
        $this->assertEquals($want, $got);
        //Test 3
        $want = [3, 2];
        $got = $obj->getQuoAndRest( -7, -3 );
        $this->assertEquals($want, $got);
    }

}

File: GO/TeoremaDaDivisaoEuclidiana_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import (
	"reflect"
	"testing"
)

func TestTeoremaDaDivisaoEuclidiana(t *testing.T) {
	tables := []struct {
		A    int
		B    int
		want []int
	}{
		{7, 3, []int{2, 1}},
		{7, -3, []int{-2, 1}},
		{-7, -3, []int{3, 2}},
	}

	for _, table := range tables {
		got := getQuoAndRest(table.A, table.B)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %v but the return was this: %v", table.want, got)
		}
	}
}
